crates/matrix-sdk/src/sliding_sync/mod.rs

```diff
@@ -719,7 +719,12 @@ impl SlidingSync {
             .await
         } else {
             self.send_sync_request(request, request_config, position_guard).await
-        }
+        };
+
+        // Notify a new sync was received
+        self.inner.client.inner.sync_beat.notify(usize::MAX);
+
+        result
```

Comment:
This isn't the same behavior as sync v3: the sync beat only beats if when the sync is successful there. Can you fix the behavior here, and add a corresponding test for this please?